orttraining/orttraining/test/training_api/test_runner.cc

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "cxxopts.hpp"
+#include "core/util/math.h"
+#include "core/common/common.h"
+#include "core/common/logging/logging.h"
+#include "core/common/logging/sinks/clog_sink.h"
+#include "core/providers/cpu/cpu_execution_provider.h"
+#include "core/session/environment.h"
+#include "core/session/inference_session.h"
+#include "core/framework/bfc_arena.h"
+#include "core/providers/cpu/cpu_provider_factory_creator.h"
+#include "orttraining/core/framework/tensorboard/event_writer.h"
+#include "orttraining/training_api/interfaces.h"
+
+using namespace onnxruntime;
+using namespace onnxruntime::common;
+using namespace onnxruntime::training;
+using namespace onnxruntime::training::tensorboard;
+using namespace std;
+
+#ifdef USE_CUDA
+namespace onnxruntime {
+
+std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Cuda(const OrtCUDAProviderOptions* provider_options);
+std::unique_ptr<IAllocator> CreateCUDAPinnedAllocator(int16_t device_id, const char* name);
+
+}  // namespace onnxruntime
+#endif
+
+static SessionOptions session_options = {
+    ExecutionMode::ORT_SEQUENTIAL,     // execution_mode
+    ExecutionOrder::PRIORITY_BASED,    // execution_order
+    false,                             // enable_profiling
+    ORT_TSTR(""),                      // optimized_model_filepath
+    true,                              // enable_mem_pattern
+    true,                              // enable_mem_reuse
+    true,                              // enable_cpu_mem_arena
+    ORT_TSTR("onnxruntime_profile_"),  // profile_file_prefix
+    "",                                // session_logid
+    -1,                                // session_log_severity_level
+    0,                                 // session_log_verbosity_level
+    5,                                 // max_num_graph_transformation_steps
+    TransformerLevel::Level1,          // graph_optimization_level
+    {},                                // intra_op_param
+    {},                                // inter_op_param
+    {},                                // free_dimension_overrides
+    true,                              // use_per_session_threads
+    true,                              // thread_pool_allow_spinning
+    false,                             // use_deterministic_compute
+    {},                                // config_options
+    {},                                // initializers_to_share_map
+};
+
+namespace onnxruntime {
+namespace training {
+namespace api_test {
+
+struct Parameters {
```

Comment:
Since we are planning to expose Parameters, change this to TestParams or Test Parameters just to avoid confusion.